docs/usage.rst

```diff
@@ -162,6 +162,69 @@ mapping that is checked against the :ref:`schema <schema_dict-rule>` rule:
    ``allow_unknown`` can also be set to a validation schema.
 
 
+.. _requiring-all:
+
+Requiring all
+-------------
+See also :ref:`this paragraph <require_all>`, :ref:`this paragraph <required>`.
+
+By default any keys defined in the schema are not required:
+
+.. doctest::
+
+    >>> schema = {'name': {'type': 'string', 'maxlength': 10}}
+    >>> v.validate({}, schema)
+    True
+
+However, you can require all document keys pairs by setting
+``require_all`` to ``True``:
+
+.. doctest::
+
+    >>> v.schema = {'name': {'type': 'string', 'maxlength': 10}}
+    >>> v.require_all = True
+    >>> v.validate({})
+    False
+
+``require_all`` can also be set at initialization:
+
+.. doctest::
+
+    >>> v = Validator({'name': {'type': 'string', 'maxlength': 10}}, require_all=True)
+    >>> v.validate({})
+    False
+    >>> v.require_all = False
+    >>> v.validate({})
+    True
+
+``require_all`` can also be set as rule to configure a validator for a nested
```

Comment:
i'm not sure, but i think we commonly use the term subdocument instead of 'nested mapping' through the docs.